redis/src/main/scala/zio/redis/TestExecutor.scala

```diff
@@ -2943,6 +2943,236 @@ private[redis] final class TestExecutor private (
+                                  }
+                                }
+                              case BitFieldType.SignedInt(size) =>
+                                val bitField = getBitField(string, offset, size)
+
+                                STM.fromOption {
+                                  parseSignedLong(bitField).map { long =>
+                                    parseValue(value, bitFieldType, overflow).map { value =>
+                                      (setBitField(string, offset, size, value), respChunk :+ RespValue.Integer(long))
+                                    }.getOrElse((string, respChunk :+ RespValue.NullBulkString))
+                                  }
+                                }
+                            }
+                          }
+              } yield newResp
+
+              respOption match {
+                case Some(resp) => runCommands(tail, overflow, resp)
+                case None       => None
+              }
+            case "INCRBY" :: encoding :: offset :: increment :: tail =>
+              val respOption = for {
+                bitFieldType <- decodeBitFieldType(encoding)
+                offset       <- decodeOffset(offset)
+                increment    <- toLongOption(increment)
+                newResp = resp.flatMap { case (string, respChunk) =>
+                            bitFieldType match {
+                              case BitFieldType.UnsignedInt(size) =>
+                                val bitField = getBitField(string, offset, size)
+
+                                STM.fromOption {
+                                  parseUnsignedLong(bitField).map { long =>
+                                    parseValue(long + increment, bitFieldType, overflow).map { value =>
+                                      (setBitField(string, offset, size, value), respChunk :+ RespValue.Integer(value))
+                                    }.getOrElse((string, respChunk :+ RespValue.NullBulkString))
+                                  }
+                                }
+                              case BitFieldType.SignedInt(size) =>
+                                val bitField = getBitField(string, offset, size)
+
+                                STM.fromOption {
+                                  parseUnsignedLong(bitField).map { long =>
+                                    parseValue(long + increment, bitFieldType, overflow).map { value =>
+                                      (setBitField(string, offset, size, value), respChunk :+ RespValue.Integer(value))
+                                    }.getOrElse((string, respChunk :+ RespValue.NullBulkString))
+                                  }
+                                }
+                            }
+                          }
+              } yield newResp
+
+              respOption match {
+                case Some(resp) => runCommands(tail, overflow, resp)
+                case None       => None
```

Comment:
This is a `map`.

Reply:
I wanted to make the function tail-recursive.